File: ruby/elements/blurb.rb
require_relative 'image'

module Elements
  class Blurb < Element
    attr_reader :text

    def initialize( compiler, article, lines)
      @text = lines.join( ' ')
      if SPECIAL_CHARS =~ @text
        article.error( 'Bad characters in blurb')
        @discard = true
      end
      if @text.size > 50
        article.error( 'Blurb too long')
        @discard = true
      end
    end

    def page_content?
      false
    end

    def special?
      true
    end
  end
end

File: java/src/com/alofmethbin/jadawin/elements/Blurb.java
package com.alofmethbin.jadawin.elements;

import com.alofmethbin.jadawin.Article;
import java.util.List;

public class Blurb extends Element {
    private String text;
    
    public Blurb( Article article, List<String> lines) {
        super( article);
        
        StringBuilder b = new StringBuilder();
        for (String line: lines) {
            if (b.length() > 0) {b.append( " ");}
            b.append( line);
        }
        
        text = checkSpecialChars( b.toString());
        
        if (text.length() > 50) {
            error( "Blurb too long");
        }
    }

    @Override
    public boolean isContent() {
        return false;
    }

    @Override
    public boolean isSpecial() {
        return true;
    }
    
    public String text() {
        return text;
    }
}
